Tests/SwiftParserTest/AttributeTests.swift

```diff
@@ -73,41 +73,66 @@ final class AttributeTests: XCTestCase {
   func testMissingClosingParenToAttribute() {
     assertParse(
       """
-      @_specializeℹ️(e1️⃣
+      @_specializeℹ️(1️⃣e2️⃣
       """,
       diagnostics: [
         DiagnosticSpec(
+          locationMarker: "1️⃣",
+          message:
+            "unknown parameter 'e' in attribute argument; valid parameters 'target', 'availability', 'exported', 'kind', 'spi', 'spiModule', and 'available'"
+        ),
+        DiagnosticSpec(
+          locationMarker: "2️⃣",
           message: "expected ':' in attribute argument",
           fixIts: ["insert ':'"]
         ),
         DiagnosticSpec(
+          locationMarker: "2️⃣",
           message: "expected ')' to end attribute",
           notes: [NoteSpec(message: "to match this opening '('")],
           fixIts: ["insert ')'"]
         ),
         DiagnosticSpec(
+          locationMarker: "2️⃣",
           message: "expected declaration after attribute",
           fixIts: ["insert declaration"]
         ),
       ],
       fixedSource: """
-        @_specialize(e:) <#declaration#>
+        @_specialize(e  <#identifier#>:) <#declaration#>
```

Comment:
Which Fix-It inserts `<#identifier#>`? The fixed source looks quite incorrect.

Reply:
It is because of this error: https://github.com/apple/swift-syntax/pull/1727/files#r1221798316

We can handle it better by etc add a Fix-It for each label that is valid? 
What do you think of that?

Otherwise we should stop parsing the line at that point?

Reply:
But which diagnostic is this attached to? No diagnostic mentions: “insert identifier” or anything of the kind. I would expect the “unknown parameter 'e'” diagnostic to remove `e` and substitute it with `<#identifier#>`.

Reply:
It's because we add a `RawTokenSyntax(missing: .identifier, arena: self.arena)`, which is wrong. 
But I could not decide how we can add a missing token, that can be one of x cases, like we can here.

Reply:
But which diagnostic does insert it? In the test case, I see the following Fix-It messages? We have 
- `insert ':'`
- `insert ')'` -> doesn’t mention the insertion of an identifier
- `insert declaration`
None of these mention the insertion of an identifier, so I’m surprised that one gets added. Based on locations, I would suspect that `insert ':'` adds the identifier but then I’m surprised that it doesn’t say `insert identifier and ':'`.